inference/engine.go

```diff
@@ -325,6 +325,11 @@ func (e *Engine) observeSiteExplanation(site primitiveSite, siteExplained Explai
 		}
 		e.conflicts.addConflict(newOverconstrainedConflict(site, trueExplanation, falseExplanation))
 
+		// Even though we have a conflict, we still need to make sure to activate any controlled
+		// triggers that are waiting on this site, so that we would not miss processing any
+		// triggers.
+		e.activateControlledTriggers(site, siteExplained)
+
```

Comment:
Not saying we shouldn't do this, but just want to make sure this is a bugfix for the inference mode, right? (since this PR is on no infer mode).

Reply:
Yes you are right; it is a bugfix for inference mode! I don't want to open a PR for this single line, since this PR with its predecessors (#8 #24) is in a stack anyway.